apicode/newapi.py

```diff
@@ -0,0 +1,76 @@
+import requests
+import inquirer
+import textwrap
+import os
+from inquirer.themes import GreenPassion
+
+
+class NewsApi:
+    # function for picking the news source from the user
+    def get_user_source(self):
+        questions = [
+            inquirer.List('sources',
+                          message="From which news source would you like to acquire your headlines?",
+                          choices=['nbc-news', 'cnbc',
+                                   'bbc-news', 'cnn'],
+                          ),
+
+
+        ]
+        # theme adds a green background to the items in the list
+        answers = inquirer.prompt(questions, theme=GreenPassion())
+        print(answers['sources'])
+
+        return answers['sources']
+    # function to retrieve a product key for the environment variable
+
+    def add_api_key(self):
+        if('API_KEY' in os.environ) == False:
+            raise ValueError(
+                "Access Denied ..., you dont have a product key in your environment variables")
+        return os.environ.get('API_KEY')
+
+    # function for collecting data for from the API
+
+    def fetch_data(self, newssource):
+
+        api_key = self.add_api_key()
```

Comment:
The api key method raises an exception.
When using such a method, you should wrap it in a `try...except` block to catch any errors that might occur

Reply:
This helps prevent your app from crashing